app/models/email_address.rb

```diff
@@ -73,4 +76,17 @@ def create_fingerprints(email)
       [Pii::Fingerprinter.fingerprint(email), *Pii::Fingerprinter.previous_fingerprints(email)]
     end
   end
+
+  private
+
+  # Remove email id from all user identities
+  # when the email is destroyed.
+  def nil_identity_email_address_id
```

Comment:
Quibbling on the name, I don't really read "nil" as a verb, and technically this email could have several identities (plural) associated with it.

```suggestion
  def reset_linked_identities
```

Reply:
Much better